pkg/apis/core/validation/seed.go

```diff
@@ -161,6 +162,17 @@ func ValidateSeedSpec(seedSpec *core.SeedSpec, fldPath *field.Path, inTemplate b
 	}
 
 	if seedSpec.Settings != nil {
+		if seedSpec.Settings.ExcessCapacityReservation != nil {
+			for i, config := range seedSpec.Settings.ExcessCapacityReservation.Configs {
+				for resource, value := range config.Resources {
+					allErrs = append(allErrs, corevalidation.ValidateResourceQuantityValue(resource.String(), value, fldPath.Child("settings, excessCapacityReservation", "configs").Index(i).Child("resources").Child(resource.String()))...)
+				}
+				if config.NodeSelector != nil {
+					allErrs = append(allErrs, corevalidation.ValidateNodeSelector(config.NodeSelector, fldPath.Child("settings, excessCapacityReservation", "configs").Index(i).Child("nodeSelector"))...)
+				}
+				allErrs = append(allErrs, corevalidation.ValidateTolerations(config.Tolerations, fldPath.Child("settings, excessCapacityReservation", "configs").Index(i).Child("tolerations"))...)
+			}
+		}
```

Comment:
You could add unit tests.

Reply:
good idea 👍 
While adding some tests I created an additional validation. It does not make sense to allow reservations which do not reserve any resources.

Additionally, I copied some tests from Kubernetes upstream for the validation of tolerations.